samples/net/cellular_modem/src/at_shell.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/modem/pipelink.h>
+
+MODEM_PIPELINK_DT_DECLARE(DT_ALIAS(modem), shell);
+
+static struct modem_pipelink *shell_pipelink = MODEM_PIPELINK_DT_GET(DT_ALIAS(modem), shell);
+static struct k_work receive_work;
+
+static uint8_t transmit_buf[255];
+static uint8_t receive_buf[255];
+
+static void receive_work_handler(struct k_work *work)
+{
+	int ret;
+	struct modem_pipe *pipe;
+
+	pipe = modem_pipelink_get_pipe(shell_pipelink);
+	if (pipe == NULL) {
+		printk("Nearly exploded, why is pipe NULL?");
+		return;
+	}
+
+	ret = modem_pipe_receive(pipe, receive_buf, sizeof(receive_buf));
+	if (ret < 1 || ret == sizeof(receive_buf)) {
+		return;
+	}
+
+	printk("%s", receive_buf);
+}
+
+static void at_shell_pipe_callback(struct modem_pipe *pipe,
+				   enum modem_pipe_event event,
+				   void *user_data)
+{
+	switch (event) {
+	case MODEM_PIPE_EVENT_OPENED:
+		printk("AT shell opened\n");
+		break;
+
+	case MODEM_PIPE_EVENT_RECEIVE_READY:
+		k_work_submit(&receive_work);
+		break;
+
+	case MODEM_PIPE_EVENT_CLOSED:
+		printk("AT shell closed\n");
+		break;
+
+	default:
+		break;
+	}
+}
+
+void pipelink_shell_callback(struct modem_pipelink *link, enum modem_pipelink_event event,
+			     void *user_data)
+{
+	ARG_UNUSED(user_data);
+
+	switch (event) {
+	case MODEM_PIPELINK_EVENT_CONNECTED:
+		modem_pipe_attach(modem_pipelink_get_pipe(link), at_shell_pipe_callback, NULL);
+		modem_pipe_open_async(modem_pipelink_get_pipe(link));
+		printk("AT shell connected\n");
+		break;
+
+	case MODEM_PIPELINK_EVENT_DISCONNECTED:
+		printk("AT shell disconnected\n");
+		break;
+
+	default:
+		break;
+	}
+}
+
+int at_shell_init(void)
+{
+	k_work_init(&receive_work, receive_work_handler);
+	modem_pipelink_attach(shell_pipelink, pipelink_shell_callback, NULL);
+	return 0;
+}
+
+SYS_INIT(at_shell_init, POST_KERNEL, 99);
+
+static int at_shell_at_cmd_handler(const struct shell *sh, size_t argc, char **argv)
+{
+	struct modem_pipe *pipe;
+
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	if (!modem_pipelink_is_connected(shell_pipelink)) {
+		shell_print(sh, "Pipe is not connected, is modem powered on?");
+		return -ENODEV;
+	}
+
+	pipe = modem_pipelink_get_pipe(shell_pipelink);
+	if (pipe == NULL) {
+		shell_print(sh, "Nearly exploded, why is pipe NULL?");
+		return -ENODEV;
+	}
+
+	snprintk(transmit_buf, sizeof(transmit_buf), "%s\r\n", argv[1]);
+
+	if (modem_pipe_transmit(pipe, transmit_buf, strlen(transmit_buf)) < 0) {
+		shell_print(sh, "Failed to transmit, is pipe open?");
+		return -EAGAIN;
+	}
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	sub_at_shell_cmds,
+	SHELL_CMD_ARG(at, NULL, "Send AT command", at_shell_at_cmd_handler, 2, 0),
+	SHELL_SUBCMD_SET_END
+);
+
+SHELL_CMD_REGISTER(at_shell, &sub_at_shell_cmds, "Modem AT shell commands", NULL);
```

Comment:
Is it possible to make it so that, instead of the command being `at_shell at AT+MYCOMMAND`, it is directly `at[_shell] AT+MYCOMMAND`?